druid/src/widget/list.rs

```diff
@@ -141,19 +141,20 @@ impl<C: Data, T: ListIter<C>> Widget<T> for List<C> {
             }
         });
 
-        let len = self.children.len();
-        if len > data.data_len() {
-            self.children.truncate(data.data_len())
-        } else if len < data.data_len() {
-            data.for_each(|child_data, i| {
-                if i < len {
-                    return;
-                }
-
-                let mut child = WidgetPod::new((self.closure)());
-                child.update(ctx, child_data, env);
-                self.children.push(child);
-            });
+        match self.children.len() {
+            len if len > data.data_len() => self.children.truncate(data.data_len()),
```

Comment:
As I said in person, I think the clippy lint is wrong here, and I would push back against it. That said, this isn't *horrible.*